rest-api/src/main/java/gov/cms/qpp/conversion/api/controllers/v1/ExceptionHandlerControllerV1.java

```diff
@@ -82,6 +84,16 @@ public ExceptionHandlerControllerV1(final AuditService auditService) {
 		return new ResponseEntity<>(exception.getMessage(), httpHeaders, HttpStatus.NOT_FOUND);
 	}
 
+	@ExceptionHandler(AmazonS3Exception.class)
+	@ResponseBody
+	ResponseEntity<String> handleGenericS3Exception(AmazonS3Exception exception) {
```

Comment:
I guess we could do this, but I was thinking of something simpler.  In `StorageServiceImpl#getFileByLocationId`, if `amazonS3#getObject` throws an exception, catch it and instead `throw new NoFileInDatabaseException(FILE_NOT_FOUND)`.

What do you think of that?

Reply:
Would that be accurate enough though? Because dynamo and s3 are different even if we use both for storage.

Reply:
I think separate the two out would be more accurate imo.

Reply:
I would like to ensure that we propagate the correct error from S3 and DynamoDB.  If a request to S3 returns a 404, for instance, that 404 should be propagated as the response code here.  I don't want to obscure the downstream error code by wrapping everything in a 400 error.

Reply:
Working on this now @sfradkin